core/src/batch/group_by.rs

```diff
@@ -83,6 +83,7 @@ where
     bridge: Bridge<B::Item>,
     groups: HashMap<D, Box<dyn Batch<Item = B::Item>>>,
     catchall: Box<dyn Batch<Item = B::Item>>,
+    fanouts: VecDeque<Disposition<B::Item>>,
```

Comment:
had the q about vecdeque here, but preserving order here is good (and for a limited number of fanouts, the data structure is pretty optimized).